### PR TITLE
Add rotate and algorithms implemented in terms of it

### PR DESCRIPTION
**Pull request by odinthenerd**

### CODE REVIEW

**Comment by odinthenerd** [comment]
not sure if erase should be called erase_at for clarity, same for insert. I'm also not sure if it is better to have a raw number int eh public interface or an int_. Open for opinions.

**Comment by chieltbest** [comment]
I think it would be better to use raw numerals in the interface, as that is more restrictive, and users might have other ways of obtaining numbers, in which case they would have to repack the numbers into a `mpl::uint_`

**Comment by odinthenerd** [comment]
agreed, but I still think types rather than numbers is best for continuations because its more generic and we do build algorithms with algorithms there where having things that are not types in the public interface can be a deal breaker

File: src/kvasir/mpl/algorithm/replace_if.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+				};
+				template<template <typename...> class F, typename Input>
+				struct replace_if_pred<lambda<F>,Input> {
+					template <typename T>
+					using f = typename conditional<F<T>::value>::template f<Input, T>;
+				};
+			}
+
+			template <typename F, typename Input>
+			using replace_if = transform<detail::replace_if_pred<F,Input>, listify>;
```
Comment (chieltbest) [inline src/kvasir/mpl/algorithm/replace_if.hpp]: replace_if has no continuation

**Pull request merged**